testing/cli/test_sct_deepseg.py

```diff
@@ -38,24 +37,67 @@ def test_model_dict():
         assert 'default' in value
 
 
-@pytest.mark.parametrize('fname_image, fname_seg_manual, fname_out, task', [
+@pytest.mark.parametrize('fname_image, fname_seg_manual, fname_out, task, thr', [
     (sct_test_path('t2s', 't2s.nii.gz'),
      sct_test_path('t2s', 't2s_seg-deepseg.nii.gz'),
      't2s_seg_deepseg.nii.gz',
-     'seg_sc_t2star'),
+     'seg_sc_t2star',
+     0.9),
+    (sct_test_path('t2', 't2.nii.gz'),
+     sct_test_path('t2', 't2_seg-manual.nii.gz'),
+     't2_seg_deepseg.nii.gz',
+     'seg_sc_contrast_agnostic',
+     0.5),
+    (sct_test_path('t2', 't2.nii.gz'),
+     None,  # TODO: implement integrity test (for now, just checking if output segmentation file exists)
+     't2_seg_deepseg.nii.gz',
+     'seg_spinal_rootlets_t2w',
+     0.5),
 ])
-def test_segment_nifti(fname_image, fname_seg_manual, fname_out, task,
+def test_segment_nifti(fname_image, fname_seg_manual, fname_out, task, thr,
                        tmp_path):
     """
     Uses the locally-installed sct_testing_data
     """
     # Ignore warnings from ivadomed model source code changing
     warnings.filterwarnings("ignore", category=SourceChangeWarning)
     fname_out = str(tmp_path/fname_out)  # tmp_path for automatic cleanup
-    sct_deepseg.main(['-i', fname_image, '-task', task, '-o', fname_out, '-thr', str(0.9)])
-    # TODO: implement integrity test (for now, just checking if output segmentation file exists)
+    sct_deepseg.main(['-i', fname_image, '-task', task, '-o', fname_out, '-thr', str(thr)])
     # Make sure output file exists
     assert os.path.isfile(fname_out)
-    # Compare with ground-truth segmentation
-    assert np.all(nibabel.load(fname_out).get_fdata() ==
-                  nibabel.load(fname_seg_manual).get_fdata()[..., 0])
+    # Compare with ground-truth segmentation if provided
+    if fname_seg_manual:
+        im_seg = Image(fname_out)
+        im_seg_manual = Image(fname_seg_manual)
+        im_seg_manual.data = im_seg_manual.data.squeeze()  # FIXME: t2s_seg-deepseg.nii.gz has an extra axis
+        dice_segmentation = compute_dice(im_seg, im_seg_manual, mode='3d', zboundaries=False)
+        assert dice_segmentation > 0.95
+
+
+@pytest.mark.parametrize('fname_image, fnames_seg_manual, fname_out, suffixes, task, thr', [
+    (sct_test_path('t2', 't2.nii.gz'),
+     [None, None],  # TODO: implement integrity test (for now, just checking if output segmentation file exists)
+     't2_deepseg.nii.gz',
+     ["_sc_seg", "_lesion_seg"],
+     'seg_sc_lesion_t2w_sci',
```

Comment:
Ditto about the SCI test -- don't have good ground truth for comparison, I think?

Reply:
[very new to the `testing` part of writing open-source code here] -- what kind of GT do you need? If it is a file that should be publicly available then i'm afraid we don't have one. We're using private datasets for SCI.

Reply:
It should be the "expected" output of running the model on `sct_testing_data/t2/t2.nii.gz` (or another image file distributed with SCT). It doesn't have to be confidential patient data, or generated by an expert, just something to check that the model is doing roughly the expected thing.

At worst, if we have at least one public image that we can use as input (with a non-zero output), then we can:
* Run the model once on the image,
* Visually check that the output looks OK,
* Save the output as our "ground truth".

That way, we at least protect against future changes that would make a large unexpected change.

Reply:
Right, thanks for the info! I can suggest one subject for testing the `_sc_seg` part of the SCI model. 

So, `sub-tokyoIngenia02` in [spine-generic data-multi-subject](https://github.com/spine-generic/data-multi-subject/tree/master/sub-tokyoIngenia02), has DCM with severe spinal cord compression. The notes on this subject can be found by running `cat participants.tsv | grep DCM` after downloading the dataset.

As DCM is also SCI (except that it is non-traumatic, whereas the model we trained on were patients with traumatic SCI), I think this subject can still be used for testing purposes.

Reply:
Thank you for the suggestion!

On second thought, I am realizing that the SCI model actually does an OK job at segmenting the spinal cord even on non-SCI images. I tested the already-present `sct_testing_data/t2.nii.gz` file:

<details>
<summary> Screenshot </summary>

![image](https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/16181459/d192c6c1-0777-4c10-af34-53adf1982e87)

</details>

Given that the existing testing data produces an adequate result, I'm wondering if the existing file is good enough, then? 

((Since, as far as `sct_deepseg` is concerned, it's the wrapper that matters, with the deepseg models being more or less black boxes that get swapped in and out. In other words, the small details of the exact segmentation we use for testing don't matter as much, and we only really want to guard against "large unexpected changes" (as suggested by @mguaypaq).))

Reply:
That said, regardless of which image we choose to use for the `_sc_seg` part, we're still missing a good test image for the `_sc_lesion` part. :thinking:

Reply:
During our 2024-01-24 meeting, we discussed the difficulty in finding a proper SCI lesion image that is also not private. 

The conclusion is that we would try to find a healthy control image with some sort of hyperintensity that would be detected as a false positive lesion, thus allowing us to test the lesion segmentation aspect. And, thankfully, @valosekj was able to find such a subject!

I can try to crop around this hyperintensity (as to keep `sct_testing_data` small) and then update https://github.com/spinalcordtoolbox/sct_testing_data/pull/12 accordingly.

Reply:
I've added the cropped image to the `sct_testing_data` PR here: https://github.com/spinalcordtoolbox/sct_testing_data/pull/12/commits/53217ad0ce1c1e3b251e6798669678f91a6874c7

I have not added any ground truth segmentations, though, because I want to wait to see the output of the new SCI model on this cropped image.

Reply:
I ran the new SCI model on the cropped image, and the output lesion seg was empty. However, when I ran the new SCI model on the uncropped image, it detected the hyperintensity just fine.

So, there must be some issue with the way I'm cropping things... :thinking:

Reply:
I remember we discussed whether to crop the images or not before training the SCIseg model and we decided to use uncropped images anticipating that the user could simply input the raw/uncropped images and they would get the result. Hence, it is kind of not surprising the model fails to predict something in cropped images!

Reply:
I wasn't able to get the "real" hyperintensity (for the cropped image) to be segmented by the SCIseg model. 

So, I tried creating a crude, unrealistic fake lesion inspired by [this figure](https://private-user-images.githubusercontent.com/53445351/294257130-e858bde3-1b6e-4adb-8897-a6d323b64c0e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc1MDYyODcsIm5iZiI6MTcwNzUwNTk4NywicGF0aCI6Ii81MzQ0NTM1MS8yOTQyNTcxMzAtZTg1OGJkZTMtMWI2ZS00YWRiLTg4OTctYTZkMzIzYjY0YzBlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA5VDE5MTMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkNTZhODVhY2JmZWFjNmUxZGJmOWMwNDA3YTUxYjg2MTI3ZDM2MTIxNTczZmU4YzU2ZGJiNDA3YzU2MmM4NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.L-0GsoCcbVqOUuDjEFz3LVhP2jQPeYeHoHWcNkMOmqU), as suggested by Julien in this week's meeting. It took a few tries to get the right size/intensity, but I finally was able to get some nonzero output from SCIseg:

![image](https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/16181459/6bf4491f-3b2f-49d7-af3f-5b49ad3398cf)

Test was amended in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4345/commits/e3a5d5054431df0b23026a1f5f81c1b70faaa6bf